java/impl/aws/src/main/java/co/worklytics/psoxy/S3Handler.java

```diff
@@ -71,21 +72,18 @@ StorageEventResponse process(String importBucket, String sourceKey, StorageHandl
             return null;
         }
 
+        S3Object s3Object = s3Client.getObject(new GetObjectRequest(importBucket, sourceKey));
         try (InputStream objectData = s3Object.getObjectContent();
              BOMInputStream is = new BOMInputStream(objectData);
-             InputStreamReader reader = new InputStreamReader(is, StandardCharsets.UTF_8)) {
+             InputStreamReader reader = new InputStreamReader(is, StandardCharsets.UTF_8);
+             PipedOutputStream outputStream = new PipedOutputStream();
+             OutputStreamWriter writer = new OutputStreamWriter(outputStream);
+             InputStream processedStream = new PipedInputStream(outputStream)) {
 
-            StorageEventRequest request = storageHandler.buildRequest(reader, importBucket, sourceKey, transform);
+            StorageEventRequest request = storageHandler.buildRequest(reader, writer, importBucket, sourceKey, transform);
 
             storageEventResponse = storageHandler.handle(request, transform.getRules());
```

Comment:
I might be totally wrong, IO in Java is always confusing:
I feel that this should be invoked after `s3Client.putObject` (obviously can't because it uses the destination objects)
but if handle does a flush it seems data will be held in memory until `putObject` starts processing, and might OOO?Will held the `writer` and `processedStream` until L98 flush?

Reply:
i agree with you, yeah ....